hack/machineset.sh

```diff
@@ -135,7 +155,26 @@ get_azure_ms() {
   local region=$2
   local az=$3
   local provider=$4
-
+  
+  # get proper SKU for platform-compatible image
+  sku=""
+  case "${WIN_SRV,,}" in
+    2022)
+      # has containers GA: https://docs.microsoft.com/en-us/windows-server/get-started/editions-comparison-windows-server-2022
+      sku="2022-datacenter-core"
+    ;;
+    2019)
+      sku="2019-Datacenter-with-Containers"
+    ;;
+    20h2)
+      echo "Note: Windows Server ${WIN_SRV} is not supported for platform ${provider}"
+      sku="datacenter-core-20h2-with-containers-smalldisk"
+    ;;
+    *)
+      error-exit "Windows Server version (1st parameter) must be \"2022\", \"2019\", or \"20h2\" (case insensitive)."
```

Comment:
I don't see the suggestion as super beneficial. The function would have to re-check the cloud provider which is already known at these call sites, so it would be bunch of unnecessary nested clauses. Modularization just to avoid the repeating the error message does more harm than good.